src/prefect/engine/results/constant_result.py

```diff
@@ -16,37 +16,37 @@ def __init__(self, value: Any = None, **kwargs: Any) -> None:
         self.value = value
         super().__init__(value=value, **kwargs)
 
-    def read(self, loc: str = None) -> Result:
+    def read(self, filepath: str) -> Result:
         """
         Returns the underlying value regardless of the argument passed.
 
         Args:
-            - loc (str): an unused argument
+            - filepath (str): an unused argument
         """
         return self
 
-    def write(self, **kwargs: Any) -> Result:
+    def write(self, value: Any, **kwargs: Any) -> Result:
         """
         Returns the repr of the underlying value, purely for convenience.
 
         Args:
+            - value (Any): the new value to store on the class
```

Comment:
Is this what we want? I thought constant results values were not supposed to be writable, only provided at instantiation

Reply:
I totally agree it's weird; I struggled a little with this.  However, if I pedantically view `Result`s as interfaces to backends, the `ConstantResult` has the class instance as its backend (at least that's what I reasoned).

Luckily because of how constants are unpacked within the runners this method will never actually be called (and I also don't expect users to interact with Constant Result types either).  

Open to change it if it's just too weird.

Reply:
I think it needs to change one of two ways -- work like it used to and stay named `ConstantResult`, or work like this and get a new name. (I don't like calling it ConstantResult if we're not making any guarantees on it never changing, even though we don't expect a user to use it.) A new name makes sense if it's not actually important that its maintains its constant value, but more that it a `PythonResult` (going with our storage theme for naming). 

That being said it seems it just so happens to stay constant in pipeline logic... you say 'Luckily' for that one which gives me ookies: I feel like that means we will have to enforce that better going forward. Arguably that is easier by making the ConstantResult truly constant. Alternatively it seems would could add (unless they exist?) tests proving constants don't overwrite their original values against whatever task runner method does the constant unpacking.

Reply:
@lauralorenz How about I raise an error if the provided `value` is not equal to `self.value`?  That way it's enforced to be constant.

Reply:
Yep, that is good with me!

Reply:
@lauralorenz updated 👍